staging/src/k8s.io/cli-runtime/pkg/genericclioptions/resource/visitor.go

```diff
@@ -490,11 +490,13 @@ func ExpandPathsToFileVisitors(mapper *mapper, paths string, recursive bool, ext
 	return visitors, nil
 }
 
-func isKustomizationDir(path string) bool {
-	if _, err := os.Stat(filepath.Join(path, constants.KustomizationFileName)); err == nil {
-		return true
+func isKustomizationFile(path string) bool {
+	fSys := fs.MakeRealFS()
+	ldr, err := loader.NewLoader(filepath.Dir(path), fSys)
+	if err != nil {
+		return false
 	}
-	return false
+	return target.IsKustomizationFile(ldr, filepath.Base(path))
```

Comment:
Yes, it will read the input twice. I wasn't aware of the use case `-f <(command)`. Will think about a way not to break it.

Reply:
I added the code of checking the mode of the file and only read it if it's a regular file.
```
    mode := fi.Mode()
	if !mode.IsRegular() {
		return false
	}
```
Then for the regular file, it will be read twice. Otherwise, it is not a kustomization file. Thus this function won't read it.

Reply:
> Then for the regular file, it will be read twice. Otherwise, it is not a kustomization file. Thus this function won't read it.

The `Check if the target is a kuztomization file by GVK, not filename` follow-up means that you need to read the content to know if the file is a kustomize file, so checking these differently based on mode doesn't seem correct.

Reply:
Kustomize reads kustomization file, but not only that. It also loads resources from relative paths, which are declared in the kustomization file. For the file content got from `kubectl create -f <(command producing manifest file output)`, the directory information is missing (or undefined). At least it is not a good way to use kustomize.